src/components/attendee/RetunableMediaElement/RetunableMediaElement.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useMemo, useState } from "react";
+
+import { AnyVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import { MediaElement } from "../MediaElement";
+import { useVideoComms } from "../VideoComms/hooks";
+import { VideoSource } from "../VideoComms/types";
+
+import { TuneBanner } from "./components/TuneBanner";
+import { Tuner } from "./components/Tuner";
+import { useRetunableMediaElement } from "./hooks";
+import { RetunableMediaSource } from "./types";
+
+import styles from "./RetunableMediaElement.module.scss";
+
+interface RetunableMediaElementProps {
+  space: WithId<AnyVenue>;
+}
+
+export const RetunableMediaElement: React.FC<RetunableMediaElementProps> = ({
+  space,
+}: RetunableMediaElementProps) => {
+  const [isTuning, setIsTuning] = useState(false);
+  const startTuning = useCallback(() => setIsTuning(true), [setIsTuning]);
+  const stopTuning = useCallback(() => setIsTuning(false), [setIsTuning]);
+  const { isLoading, settings } = useRetunableMediaElement({
+    spaceId: space.id,
+  });
+  const { localParticipant, remoteParticipants } = useVideoComms();
+
+  const mediaElement = useMemo(() => {
+    switch (settings.sourceType) {
```

Comment:
Generally, I agree with the map for factories. However, in this case we're using the type information from `settings.sourceType` to allow each case block to use the more specific type. I can't immediately see a way of getting Typescript to give us this in a map (I tried). I'll make a comment that this could be improved at a later date.